accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,395 @@
+
+This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used. -->
+
+## Generic design: integrate Uyuni/SUSE Manager with a cluster provider
+
+### Deploy a cluster using Uyuni/SUSE Manager
+
+Under Uyuni/SUSE Manager main menu, a new menu item "Clusters" should be introduced.
+All deployable and manageable cluster products (e.g. CaaSP, SOC, SES) will be in the sub-menu.
+
+Under that item, it should be possible to create, edit, and remove a cluster.
+
+To create a cluster, a user has to enter its name.
+
+Every detail about the cluster is saved into corresponding database tables (e.g. `susecluster`).
+
+API support for all the activities described below should be implemented as well as front-end support.
+
+#### Cluster requirements
+
+If the cluster required a specific set of nodes optionally with a defined configuration, the cluster provider must offer a way to deploy the whole infrastructure automatically, e.g. using AutoYaST profiles.
+
+Uyuni/SUSE Manager will provision bare-metal machines using the supplied AutoYaST profile to have a complete set of machines ready to be hosting the cluster.
+
+#### The Management Node
+
+The cluster provider must also indicate:
+
+- how and the number of Management Node(s) must be elected among the provisioned machines (e.g. based on hardware requirements)
+- how Management Node(s) should be deployed (e.g. additional software to be installed via AutoYaST or Salt states)
+- whether Management Node(s) can be managed by Uyuni/SUSE Manager (specifically: can users manage the updates of the Management Node via Uyuni/SUSE Manager?)
+
+#### Cluster bootstrapping
+
+The cluster must be initiated by the Management Node that has all the tools to bootstrap and manage the cluster. Specific instructions on how to bootstrap the cluster must be provided.
+Questions that need to be answered by the cluster provider:
+- Upon successful bootstrapping, can the worker nodes be managed by Uyuni/SUSE Manager?
+- Do the nodes have a specific agent running on them and/or can we install a `salt-minion` or use `salt-ssh` on those nodes?
+- Can the nodes be managed by Uyuni/SUSE Manager? Can the packages be updated or do they require specific handling by the Management Node?
+
+### Manage a cluster using Uyuni/SUSE Manager
+
+Uyuni/SUSE Manager offers to show the status of the running cluster by either accessing the cluster directly or via the Management Node.
+In this regard, the cluster provider should provide a way to expose the cluster status (e.g. running nodes, the health of each node, monitoring metrics...) to Uyuni/SUSE Manager.
+
```

Comment:
If the corrective action for restoring green health can be issued by SUSE Manager I think it is a good idea to show the health status, what do you think?

Reply:
I think restore green health cannot be issued by SUSE Manager in those cases. A series of manual operations need to be manually triggered from ceph dashboard. However, I expect in the future that this might change, assuming more operations of the ceph dashboard are being made "native" to rook. This I think would need confirmation from the SES team. However, next to that, I don't expect we'll have a single button "fix openstack" :)

Reply:
It might also depend on the cluster provider, but I am not imagining a button "fix openstack". Rather, I imagine that there is a descriptive text of the reason why a node is not on green health with (optionally) some action to be issued by SUSE Manager to try and restore the green health.
What do you think?

Reply:
I think @cavalheiro or @renner can help here